src/features/wallet/root/WalletView.tsx

```diff
@@ -58,50 +127,129 @@ const WalletView = ({
+            [0, insets.top],
             Extrapolate.CLAMP,
           ),
         },
       ],
     }),
-    [animatedCardIndex, layout.chartHeight],
+    [animatedCardIndex],
   )
 
-  if (activityViewState === 'no_activity') return null
+  const handleLayout = useCallback(
+    (key: 'condensedHeader' | 'header') => (event: LayoutChangeEvent) => {
+      const nextViewHeights = {
+        ...viewHeights,
+        [key]: { value: event?.nativeEvent.layout.height, hasLayout: true },
+      }
+
+      if (
+        nextViewHeights.condensedHeader.hasLayout &&
+        nextViewHeights.header.hasLayout
+      ) {
+        // We only want to call this once
+        animateTransition('WalletView.HeaderLayoutFinished')
+      }
+
+      setViewHeights(nextViewHeights)
+    },
+    [viewHeights],
+  )
+
+  const listTopPadding = useMemo(() => {
+    const spacingKey = 'l' as Spacing
+    return { key: spacingKey, value: spacing[spacingKey] }
+  }, [spacing])
+
+  const snapPoints = useMemo(() => {
+    const collapsedHeight = listTopPadding.value
+    const belowStatusBar = height - tabBarHeight - insets.top
+    const midHeight = belowStatusBar - viewHeights.header.value
+    const expandedHeight = belowStatusBar - viewHeights.condensedHeader.value
+
+    return [collapsedHeight, midHeight, expandedHeight]
+  }, [
+    height,
+    insets.top,
+    listTopPadding.value,
+    tabBarHeight,
+    viewHeights.condensedHeader.value,
+    viewHeights.header.value,
+  ])
+
+  const loading = useMemo(() => activityViewState === 'undetermined', [
+    activityViewState,
+  ])
+
   return (
     <>
-      <Box paddingHorizontal="l">
-        <BarChart height={layout.chartHeight} />
-      </Box>
-      <Animated.View style={balanceCardStyles}>
-        <BalanceCard
-          layout={layout}
-          onReceivePress={onReceivePress}
-          onSendPress={onSendPress}
+      <Animated.View style={condensedHeaderStyle}>
+        <WalletHeaderCondensed
+          onReceivePress={toggleShowReceive}
+          onSendPress={handleSendPress}
+          onLayout={handleLayout('condensedHeader')}
+          balance={balanceInfo}
         />
       </Animated.View>
+      <Box flex={1}>
+        {activityViewState === 'activity' && (
+          <Box onLayout={handleLayout('header')}>
+            <WalletHeader
+              handleScanPressed={navScan}
+              opacity={loading ? 0 : 1}
+            />
+            <BalanceCard
+              opacity={loading ? 0 : 1}
+              onReceivePress={toggleShowReceive}
+              onSendPress={handleSendPress}
+              balanceInfo={balanceInfo}
+              account={account}
+              accountLoading={
+                fetchAccountState === 'idle' || fetchAccountState === 'rejected'
+              }
+              toggleConvertHntToCurrency={toggleConvertHntToCurrency}
+            />
+          </Box>
+        )}
+        {activityViewState === 'no_activity' && <WalletEmpty />}
```

Comment:
I see this while the activity is initially loading, I think before there was a spinner. Was this an intended change?

Reply:
Yeah, that animation is intended. I guess we'll see what people think.

Reply:
Ok cool, looks good then